File: javascript/fibonacci.js
function fibonacci(num) {
  fibonacciArray = [0, 1];
  for (i = 0; i < num; i++) {
    fibonacciArray.push(
      fibonacciArray[fibonacciArray.length - 1] +
        fibonacciArray[fibonacciArray.length - 2]
    );
  }
  return fibonacciArray[num];
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
def fibonacci(num)
  fibonacci_array = [0, 1]
  (num - 1).times do |number|
    fibonacci_array << fibonacci_array[-1] + fibonacci_array[-2]
  end
  fibonacci_array[num]

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file



# And a written explanation of your solution
